server.go

```diff
@@ -57,9 +73,10 @@ func main() {
 
 		//logs an incoming message
 		fmt.Printf("Received message %s -> %s \n", conn.RemoteAddr(), conn.LocalAddr())
-
+		fmt.Printf("Waiting %d seconds\n", delayVal)
```

Comment:
This logs the **last** `delayVal`. It should go at L110 to capture the updated value that will be used for the wait.